### PR TITLE
Revert  changes to ChibiOS Suspend Code

### PR DESCRIPTION
**Pull request by drashna**
## Description

Current changes has some issues, reverting until we can get all this properly sorted

## Types of Changes

- [x] Core
- [x] Bugfix

## Issues Fixed or Closed by This PR

* #19780 
* #21676
* #21391

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by dunk2k — APPROVED** [review]

**Comment by KarlK90** [comment]
Another thing came to my mind: As the resume from suspend on MacOS is broken for rp2040 based boards it would be good to add an empty override for `restart_usb_driver` in the board files. Could that be added here? This would carry over the fixes for rp2040 without the regressions introduced in the other PRs.

**Comment by zvecr** [comment]
> As the resume from suspend on MacOS is broken for rp2040 based boards it would be good to add an empty override for restart_usb_driver in the board files. Could that be added here?

Probably not best to mix reverts with additional workarounds.

File: tmk_core/protocol/chibios/chibios.c [hunk 1]
```diff
@@ -157,32 +180,25 @@ void protocol_post_init(void) {
 }
 
 void protocol_pre_task(void) {
+    usb_event_queue_task();
 #if !defined(NO_USB_STARTUP_CHECK)
     if (USB_DRIVER.state == USB_SUSPENDED) {
         dprintln("suspending keyboard");
         while (USB_DRIVER.state == USB_SUSPENDED) {
             suspend_power_down();
```
Comment (zvecr) [inline tmk_core/protocol/chibios/chibios.c:188]: Maybe not the most important suggestion, but these comments used to exist (and might make applying the patch back a pain)
```suggestion
            /* Do this in the suspended state */
            suspend_power_down(); // on AVR this deep sleeps for 15ms
            /* Remote wakeup */
```

File: tmk_core/protocol/chibios/chibios.c [hunk 2]
```diff
@@ -157,32 +180,25 @@ void protocol_post_init(void) {
 }
 
 void protocol_pre_task(void) {
+    usb_event_queue_task();
```
Comment (zvecr) [inline tmk_core/protocol/chibios/chibios.c:183]:
```suggestion
    usb_event_queue_task();

```

**Review by zvecr — APPROVED** [review]

**Pull request merged**